src/util/trace.h

```diff
@@ -5,10 +5,6 @@
 #ifndef BITCOIN_UTIL_TRACE_H
 #define BITCOIN_UTIL_TRACE_H
 
-#if defined(HAVE_CONFIG_H)
```

Comment:
I also really like this change, even if it's just incidental.  It's very unlikely that out-of-tree builds would be interested in USDT (though they could of course still access it by setting the define).

One less low-level header including this file.